test/testcafe/enLeaks/EnglishLeaks_spec.js

```diff
@@ -0,0 +1,164 @@
+  'authenticator-enroll-phone.json',
+  'authenticator-enroll-phone-voice.json',
+  'authenticator-enroll-ov-via-sms.json',
+  'authenticator-enroll-ov-via-email.json',
+  'authenticator-enroll-google-authenticator.json',
+  'authenticator-enroll-email.json',
+  'authenticator-enroll-data-phone.json',
+  'authenticator-enroll-data-phone-voice.json',
+  'terminal-enduser-email-consent-denied.json',
+  'terminal-return-email-consent-denied.json',
+  'terminal-return-email-consent.json',
+  'email-challenge-consent.json',
+  'error-authenticator-enroll-idp.json',
+  'error-authenticator-verification-idp.json',
+];
+
+const parseMockData = () => {
+  // parse mocks folder
+  const mocks = [];
+  // eslint-disable-next-line no-console
+  console.log('================= Parsing mocks for en leaks automation =============');
+  fs.readdirSync(mocksFolder).forEach(file => {
+    const isIgnored = ignoredMocks.includes(file);
+    if (!isIgnored) {
+      mocks.push(file);
+    } else {
+      test.skip(`Warning skipping mock ${file} from test english leaks test suite. This file may result in english leaks on the UI.`,() => {});
+    }
+  });
+  return mocks;
+};
+
+const setUpResponse = (filePath) => {
+  const mockResponse = require(filePath);
+  // Majority of the mock files can be loaded with just mocking the introspect API and providing a response
+  // In some cases we may need to mock more than just introspect API to load the screen (example polling)
+  // For those cases add the url and responses to responseMap
+  const responseMap = [
+    { 
+      'url': 'http://localhost:3000/idp/idx/introspect',
+      'response': mockResponse
+    },
+    { 
+      'url': 'http://localhost:3000/idp/idx/challenge/poll',
+      'response': mockResponse
+    },
+  ];
+
+  const mock = RequestMock();
+  for (let i = 0; i < responseMap.length; i++) {
+    const url = responseMap[i].url;
+    const response = responseMap[i].response;
+    mock
+      .onRequestTo(url)
+      .respond(response);
+  }
+  return mock;
+};
+
+async function setup(t, locale) {
+  const widgetView = new PageObject(t);
+  await widgetView.navigateToPage();
+  await renderWidgetInLocale(`${locale}`);
+}
+
+const testEnglishLeaks = (mockIdxResponse, fileName, locale) => {
+  test.requestHooks(mockIdxResponse)(`${fileName} should not have english leaks`, async t => {
+    await setup(t, locale);
+    const viewText = await Selector('#okta-sign-in').textContent;
+    const noTranslationContentExists = await Selector('.no-translate').exists;
+    const noTranslationContent = noTranslationContentExists && await Selector('.no-translate').textContent;
+    await assertNoEnglishLeaks(fileName, viewText, noTranslationContent);
+  });
+};
+
+const runTest = (mockData) => {
+  for (let i = 0; i < mockData.length; i++) {
+    const fileName = mockData[i];
+    const filePath = `${mocksFolder}/${fileName}`;
+    // start test
+    testEnglishLeaks(setUpResponse(filePath), fileName, 'ok-PL');
```

Comment:
This method is `async` - so let's make sure we're awaiting the result before proceeding in the loop.

Reply:
Are you referring to `testEnglishLeaks` being async here ?

Reply:
Yeah - maybe not that big of a deal as it's handled in the form of a callback, but am curious if these are executed in order or happen in true `async` fashion.

Reply:
The `testEnglishLeaks ` method itself is not async. Like you mentioned the async portion is in the callback that is passed into the test for which we have awaits.